scripts/docker/ubuntu.14.04/Dockerfile

```diff
@@ -0,0 +1,87 @@
+#
+# Copyright (c) .NET Foundation and contributors. All rights reserved.
+# Licensed under the MIT license. See LICENSE file in the project root for full license information.
+#
+
+# Dockerfile that creates a container suitable to build dotnet-cli
+FROM ubuntu:14.04
+
+# Misc Dependencies for build
+RUN apt-get update && apt-get -qqy install curl unzip gettext sudo
+
+# This could become a "microsoft/coreclr" image, since it just installs the dependencies for CoreCLR (and stdlib)
+RUN echo "deb http://llvm.org/apt/trusty/ llvm-toolchain-trusty-3.6 main"  | tee /etc/apt/sources.list.d/llvm.list && \
+    curl http://llvm.org/apt/llvm-snapshot.gpg.key | apt-key add - && \
+    apt-get update && apt-get -qqy install\
+	libc6 \
```

Comment:
I disagree that these dockerfiles should share implementation because I don't think we should need to reason about anything outside of the dockerfile itself to determine what the environment the docker image created from the dockerfile looks like. 

Having a shared implementation would require adding in a file to the docker container from our repo, and then executing that file with in as a part of the image build. So far we've managed to avoid needing to do that. This would add additional complexity and room for error between the different environments where the docker container is built since it's no longer self-contained.

What benefit do you see from the shared implementation?